selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/ProjectExplorer.java

```diff
@@ -655,14 +656,23 @@ public void waitRemoveItemsByPath(String pathToItem) {
    * @param path item's path in format: "Test/src/pom.xml".
    */
   public void openContextMenuByPathSelectedItem(String path) {
-    waitItem(path);
-    waitAndSelectItem(path);
-    waitItemIsSelected(path);
-
-    Actions action = actionsFactory.createAction(seleniumWebDriver);
-    action.moveToElement(waitAndGetItem(path)).contextClick().perform();
+    for (int i = 1; ; i++) {
+      waitAndSelectItem(path);
 
-    waitContextMenu();
+      actionsFactory.createAction(seleniumWebDriver).contextClick().perform();
+      waitContextMenu();
+
+      try {
+        waitItemIsSelected(path, REDRAW_UI_ELEMENTS_TIMEOUT_SEC);
+        return;
+      } catch (WebDriverException e) {
```

Comment:
`WebDriverException` looks too general here. There is, for example, *UnreachableBrowserException* type of WebDriverException after which it doesn't make sense to repeat operation again.
`TimeoutException` would be enough in case of waiting for web element.